servicenow/check.py

```diff
@@ -0,0 +1,199 @@
+# (C) Datadog, Inc. 2010-2016
+# All rights reserved
+# Licensed under Simplified BSD License (see LICENSE)
+
+# stdlib
+
+# 3rd party
+import requests
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+from requests.packages.urllib3 import disable_warnings
+
+# Disable https warnings
+disable_warnings(InsecureRequestWarning)
+
+# project
+from checks import AgentCheck, CheckException
+
+EVENT_TYPE = SOURCE_TYPE_NAME = 'servicenow'
+
+
+class ServicenowCheck(AgentCheck):
+
+    INSTANCE_TYPE = "servicenow_cmdb"
+    SERVICE_CHECK_NAME = "servicenow.cmdb.topology_information"
+    service_check_needed = True
+    service_check_done = False
+
+    auth = None  # http basic authentication
+    timeout = None  # connection timeout
+    instance_key = None  # unique key to identify this check from its instances.
+    # The same check might be run on multiple clusters
+    base_url = None  # base url of where CMDB instance can be found, for example: https://dev12406.service-now.com
+    relation_types = {}  # relation types that are available in table cmdb_rel_type
```

Comment:
As an afterthought. All these variables should not be class variables on the check but local to the check function. The data should then be passed down to functions that need that. This to promote a more functional non-side-effecting style of programming